core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1334,20 +1372,31 @@ class ReplicaManager(val config: KafkaConfig,
         }
       }.map(_.topicPartition)
 
-      info(s"Partitions ${newOfflinePartitions.mkString(",")} are offline due to failure on log directory $dir")
+      val partitionsWithOfflineFutureReplica = allPartitions.values.filter { partition =>
+        if (partition eq ReplicaManager.OfflinePartition)
+          false
+        else partition.getReplica(Request.FutureLocalReplicaId) match {
+          case Some(replica) =>
+            replica.log.isDefined && replica.log.get.dir.getParent == dir
+          case None => false
+        }
+      }
+      replicaFetcherManager.removeFetcherForPartitions(newOfflinePartitions.toSet)
+      replicaAlterDirManager.removeFetcherForPartitions(newOfflinePartitions.toSet ++ partitionsWithOfflineFutureReplica.map(_.topicPartition))
 
+      partitionsWithOfflineFutureReplica.foreach(partition => partition.removeFutureLocalReplica())
       newOfflinePartitions.foreach { topicPartition =>
         val partition = allPartitions.put(topicPartition, ReplicaManager.OfflinePartition)
         partition.removePartitionMetrics()
       }
-
       newOfflinePartitions.map(_.topic).toSet.foreach { topic: String =>
         val topicHasPartitions = allPartitions.values.exists(partition => topic == partition.topic)
         if (!topicHasPartitions)
           brokerTopicStats.removeMetrics(topic)
       }
+      info(s"Partitions ${newOfflinePartitions.mkString(",")} are offline due to failure on log directory $dir")
```

Comment:
Sure. I have updated the patch to combine these two statements and the statement in line 1401 into this:

```
      info(s"Broker $localBrokerId stopped fetcher for partitions ${newOfflinePartitions.mkString(",")} and stopped moving logs " +
           s"for partitions ${partitionsWithOfflineFutureReplica.mkString(",")} because they are in the failed log directory $dir.")
```